lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -36,13 +37,18 @@ def execute(self):
             host_tools = {}
             for group in groups:
                 host_tools[group] = {}
-                for path in self.gen_tools_group_dir(group).glob("*/**"):
-                    if self.context.with_option:
-                        host_tools[group][path.name] = [
-                            p.read_text() for p in self.tools(path)
-                        ]
-                    else:
-                        host_tools[group][path.name] = [p for p in self.tools(path)]
+                try:
+                    for path in self.gen_tools_group_dir(group).glob("*/**"):
+                        if self.context.with_option:
+                            host_tools[group][path.name] = [
+                                (p, (path / p).read_text()) for p in self.tools(path)
+                            ]
+                        else:
+                            host_tools[group][path.name] = [p for p in self.tools(path)]
+                except BadToolGroup:
+                    self.logger.error("Bad tool group: %s", group)
+                    # 0.69.9 behavior
+                    return 0
             if host_tools:
                 for k, v in host_tools.items():
                     for h, t in v.items():
```

Comment:
I think we want these traversals in sorted order, for test reproducibility.  (Ditto line 73.)

Reply:
That will be done in the subsequent PR. All I'm interested in in this PR is getting rid of the tracebacks. I don't really care about the output (yet).